mindsdb/integrations/handlers/chromadb_handler/chromadb_handler.py

```diff
@@ -310,9 +311,28 @@ def select(
             payload = {column: payload[column] for column in columns}
 
         # always include distance
+        distance_filter = None
+        distance_col = TableField.DISTANCE.value
         if distances is not None:
-            payload[TableField.DISTANCE.value] = distances
-        return pd.DataFrame(payload)
+            payload[distance_col] = distances
+
+            for cond in conditions:
+                if cond.column == distance_col:
+                    distance_filter = cond
```

Comment:
Should this `break` after finding condition?

Reply:
added